lib/kafka_producer.js

```diff
@@ -202,8 +203,16 @@ KafkaProducer.prototype.batch = function batch(topic, message, timeStamp, callba
         return;
     }
 
+    if (typeof message !== 'string' && !Buffer.isBuffer(message)) {
+        var err = new Error('For batching, message must be a string or buffer, not ' + (typeof message));
+        if (!callback) {
+            throw err;
```

Comment:
We use this same pattern in `message_batch.js`, so I was cribbing off of that. Would it be better to swallow the error? I'd expect that anyone using this library would want to know if their code was trying to send the wrong type of object to kafka rather than having it fail silently.